File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int getMaxF(vector<int>& counter){
        int count = 0;
        
        for(int i=0; i<26; i++){
            count = max(count, counter[i]);
        }
        
        return count;
    }
    int characterReplacement(string s, int k) {
        int ans = 0;
        
        vector<int> counter(26, 0);
        int li = 0;
        int ri = 0;
        int maxF = 0;
        
        while(ri < s.size()){
            counter[s[ri]-'A']++;
            maxF = max(maxF, counter[s[ri]-'A']);
            
            if((ri-li+1) - maxF > k){
                counter[s[li]-'A']--;
                li++;
            }
            
            ans = max(ans, ri-li+1);
            ri++;
            
            //cout<<ans<<" ";
        }
        
        return ans;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        l = 0
        freq = {}
        maxF = 0
        res = 0
        
        for r in range(len(s)):
            freq[s[r]] = 1 + freq.get(s[r], 0)
            maxF = max(maxF, freq[s[r]])
            
            while (r-l+1)-maxF > k:
                freq[s[l]] -= 1
                l += 1
        
        return (r-l+1)